sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -319,6 +319,50 @@ class RapidsBufferCatalog(
     makeNewHandle(id, initialSpillPriority)
   }
 
+  /**
+   * Adds a batch to the device storage. This does NOT take ownership of the
+   * batch, so it is the responsibility of the caller to close it.
+   *
+   * @param batch                batch that will be owned by the store
```

Comment:
well ok.. it "will" be owned by the store. It isn't "currently" owned as you are adding it, but once the caller closes we definitely own it. So it's definitely in the near future. 

The comment above is to make sure that we don't forget a `withResource` while calling this function.. Some other functions take the object and no `withResource` is needed (e.g. they take ownership).

Reply:
Yeah - I get what you are saying, but the comment on line 326 would make me think I don't need to use `withResource` (although the previous comment clearly indicates need to).  Maybe just say `batch that will be added to the store` for this one so it doesn't sound like a contradiction?

Reply:
So for this case will change it to "added". The reason the `Table` takes ownership is because that's the only thing it could do (because `cudf.Table` isn't a regular ref counted cuDF object).

The other add methods are all dealing with objects that are either closeable, or close is a proxy to something else that is (ColumnarBatch is one example of that).

If we make `cudf.Table` ref countable, we can then in `addTable` incRefCount the table itself, then have the caller close it, giving this a more consistent treatment. Does that sounds reasonable? If so I can file a cuDF issue for it.